platform/storage/zephyr/nvs/src/mender-storage.c

```diff
@@ -401,6 +411,41 @@ mender_storage_delete_provides(void) {
 #endif /* CONFIG_MENDER_PROVIDES_DEPENDS */
 #endif /* CONFIG_MENDER_FULL_PARSE_ARTIFACT */
 
+mender_err_t
+mender_storage_set_artifact_name(char *artifact_name) {
+
+    assert(NULL != artifact_name);
+
+    /* Write artifact_name */
+    if (!checked_nvs_write(&mender_storage_nvs_handle, MENDER_STORAGE_NVS_PROVIDES, artifact_name, strlen(artifact_name) + 1)) {
+        mender_log_error("Unable to write artifact_name");
+        free(artifact_name_str);
+        return MENDER_FAIL;
+    }
+
+    return MENDER_OK;
+}
+
+mender_err_t
+mender_storage_get_artifact_name(char **artifact_name) {
+
+    assert(NULL != artifact_name);
+    size_t artifact_name_length = 0;
+
+    /* Read artifact_name */
+    mender_err_t ret = nvs_read_alloc(&mender_storage_nvs_handle, MENDER_STORAGE_NVS_PROVIDES, (void **)artifact_name_str, &artifact_name_length);
```

Comment:
No `artifact_name_str` here.

Reply:
This one is not resolved, as far as I can see?

Reply:
You're right, guess I didn't commit it, thanks